wagtail/admin/views/generic/models.py

```diff
@@ -491,6 +491,31 @@ def setup(self, request, *args, **kwargs):
         super().setup(request, *args, **kwargs)
         self.action = self.get_action(request)
 
+    def post(self, request, *args, **kwargs):
+        # BaseCreateView.post() would set self.object here, but some subclasses need
+        # to do that during setup() instead, so only do that if it hasn't already been set.
+        if not hasattr(self, "object"):
+            self.object = None
```

Comment:
I think eventually we should get rid of `self.object = self.get_object()` as I explained in https://github.com/wagtail/wagtail/issues/11590#issuecomment-3632351690, but for now I think this is OK